src/main/java/org/openstreetmap/atlas/checks/validation/tag/TollValidationCheck.java

```diff
@@ -0,0 +1,624 @@
+                recursionCount++;
+                return this.getNearbyTollFeatureInEdgeSide(inEdge, alreadyCheckedNearbyTollEdges,
+                        recursionCount);
+            }
+        }
+        return Optional.empty();
+    }
+
+    /**
+     * @param edge
+     *            edge in question
+     * @return nearby toll feature id on the out edge side of the edge in question (downstream)
+     */
+    private Optional<Long> getNearbyTollFeatureOutEdgeSide(final Edge edge,
+            final Set<Long> alreadyCheckedNearbyTollEdges, double recursionCount)
+    {
+        final Set<Edge> outEdges = this.getOutEdges(edge);
+
+        for (final Edge outEdge : outEdges)
+        {
+            if (outEdges.size() >= this.minInAndOutEdges && this.edgeIntersectsTollFeature(outEdge)
+                    && !alreadyCheckedNearbyTollEdges.contains(outEdge.getIdentifier()))
+            {
+                return this.getAreaOrNodeIntersectionId(outEdge, alreadyCheckedNearbyTollEdges);
+            }
+            if (recursionCount <= this.maxIterationForNearbySearch
+                    && outEdges.size() >= this.minInAndOutEdges
+                    && !this.edgeIntersectsTollFeature(outEdge)
+                    && !alreadyCheckedNearbyTollEdges.contains(outEdge.getIdentifier()))
+            {
+                alreadyCheckedNearbyTollEdges.add(outEdge.getIdentifier());
+                recursionCount++;
+                return this.getNearbyTollFeatureOutEdgeSide(outEdge, alreadyCheckedNearbyTollEdges,
+                        recursionCount);
+            }
+        }
+        return Optional.empty();
+    }
+
+    /**
+     * @param edge
+     *            some edge
+     * @return out edges that are car navigable and positive (eliminates reverse edges)
+     */
+    private Set<Edge> getOutEdges(final Edge edge)
+    {
+        return edge.outEdges().stream().filter(
+                someEdge -> someEdge.isMainEdge() && HighwayTag.isCarNavigableHighway(someEdge))
+                .collect(Collectors.toSet());
+    }
+
+    /**
+     * @param edge
+     *            some edge
+     * @return tag inconsistencies between 3 consecutive edges.
+     */
+    private boolean hasInconsistentTollTag(final Edge edge)
+    {
+        final Set<Edge> inEdges = edge.inEdges().stream()
+                .filter(inEdge -> inEdge.getOsmIdentifier() != edge.getOsmIdentifier()
+                        && inEdge.isMainEdge() && HighwayTag.isCarNavigableHighway(inEdge))
+                .collect(Collectors.toSet());
+        final Set<Edge> outEdges = edge.outEdges().stream()
+                .filter(outEdge -> outEdge.getOsmIdentifier() != edge.getOsmIdentifier()
+                        && outEdge.isMainEdge() && HighwayTag.isCarNavigableHighway(outEdge))
+                .collect(Collectors.toSet());
+        if (inEdges.size() == this.minInAndOutEdges && outEdges.size() == this.minInAndOutEdges)
```

Comment:
Not a huge deal but we could rename this variable (especially in the config) as it seems to be functioning as an exact constraint, not a min

Reply:
For this specific instance I am just using the config value since it is 1. That variable is used other places as a min. I can just use a different variable.